solver/src/main/java/org/chocosolver/solver/constraints/binary/PropPowEven.java

```diff
@@ -91,14 +76,16 @@ void setBounds() throws ContradictionException {
     }
 
     int floor_nroot(int n) {
-        if (n < 0)
+        if (n < 0) {
             return 0;
+        }
         return MathUtils.safeCast((long) Math.floor(Math.pow(n, 1. / exponent)));
     }
```

Comment:
shouln't we use PRECISION here as well ?

Reply:
I don't know, I did not touch the code wrt floor_nroot and ceil_nroot, but maybe.

Reply:
in my understanding, once double are involved, we should always use PRECISION to avoid numerical issues. 
@cprudhom any idea why it is used for PropPowOdd and not PropPowEven ?

Reply:
It is interesting to note that the result of Math.pow has a precision of 1 ulp (https://docs.oracle.com/javase/6/docs/api/java/lang/Math.html#pow%28double,%20double%29), which is about 4e-16 in Java (https://en.wikipedia.org/wiki/Unit_in_the_last_place#Example_2).

If we are afraid of cases like 2.999999999 as a result instead of 3, we could always add 1e-9 for UB (and remove 1e-9 for LB). It should be OK then, do you agree ?

Reply:
I feel it would be safer yes